Update API docs (#212)
sourcebots/docs: docs/programming/index.md

# Programming

Programming your robot is done in [Python](https://www.python.org/),
specifically version 3.9.6. You can learn more about Python from their
[docs](https://docs.python.org/3/), and our whirlwind tour.

Each board has an API which allows its various functionality to be controlled.

## Setup

The following two lines are required to complete initialisation of the
kit:

``` python

> from sbot import *

Reviewer: ```suggestion
from sbot import Robot
```

robot's functions.

The remainder of the tutorials pages will assume your `Robot` object is
defined as `r`.

## Running your code

Your code needs to be put on a USB drive in a file called `robot.py`. When connected to the robot, this file will be executed. The file is directly executed off your USB drive, with your drive as the working directory.

!!! tip
    If this file is missing or incorrectly named, your robot won't do anything. No log file will be created.

To stop your code running, you can just remove the USB drive. This will also stop the motors and any other peripherals connected to the kit.

You can then reinsert the USB drive into the robot, and it will run your `main.py` again (from the start). This allows you to make changes and test them quickly.

## Logs

A log file is saved to your USB so you can see what your robot did,
what it didn't do, and any errors it raised. The file is saved to log.txt in the top-level directory of the USB drive.

!!! warning
    The previous log file is deleted at the start of each run, so copy it elsewhere if you need to keep hold of it!

## Running Code before pressing the start button

> If you want to do things before the start button press, such as setting up servos or motors, you can pass `wait_start` to the `Robot` constructor. You will then need to wait for the start button manually.

Reviewer: Could this be `wait_for_start`?

Author: https://github.com/sourcebots/sbot/pull/327

```python
r = Robot(wait_start=False)

# Do your setup here

r.wait_start()
```

## Debug mode

It is possible to run your robot in "Debug Mode".

In "Debug Mode", your robot will print more information about what it is doing.

```python
from sbot import Robot
r = Robot(debug=True)
```

!!! info

>     Debug mode is quite verbose. It will print a lot of information that you will not need.

Reviewer: nit: maybe tweak wording?
```suggestion
    Debug mode is very verbose. It will print a lot of information that you may not need.
```

## Included Libraries

> Python already comes with plenty of [built-in

Reviewer: nit: already feels unecessary
```suggestion
Python comes with plenty of [built-in
```
